src/safeposix/syscalls/fs_calls.rs

```diff
@@ -180,27 +261,35 @@ impl Cage {
         fd //open returns the opened file descriptor
     }
 
-    //------------------MKDIR SYSCALL------------------
-    // Description
-    // The mkdir_syscall() creates a new directory named by the path name pointed to by a path as the input parameter in the function.
-    // The mode of the new directory is initialized from the "mode" provided as the input parameter in the function.
-    // The newly created directory is empty with size 0 and is associated with a new inode of type "DIR".
-    // On successful completion, the timestamps for both the newly formed directory and its parent are updated along with their linkcounts.
-    
-    // Function Arguments
-    // The mkdir_syscall() receives two arguments:
-    // 1. Path - This represents the path at which the new directory will be created.
-    //           For example: "/parentdir/dir" represents the new directory name as "dir", which will be created at this path (/parentdir/dir).
-    // 2. Mode - This represents the permission of the newly created directory. 
-    //           The general mode used is "S_IRWXA": which represents the read, write, and search permissions on the new directory. 
+    /// ## ------------------MKDIR SYSCALL------------------
+    /// 
+    /// ### Description
+    /// 
+    /// The mkdir_syscall() creates a new directory named by the path name pointed to by a path as the input parameter in the function.
+    /// The mode of the new directory is initialized from the "mode" provided as the input parameter in the function.
+    /// The newly created directory is empty with size 0 and is associated with a new inode of type "DIR".
+    /// On successful completion, the timestamps for both the newly formed directory and its parent are updated along with their linkcounts.
     
-    // Return Values
-    // Upon successful creation of the directory, 0 is returned.
-    // Otherwise, an error with a proper errorNumber and errorMessage is returned based on the different scenarios.
-    //
-    // Tests
-    // All the different scenarios for mkdir_syscall() are covered and tested in the "fs_tests.rs" file under "mkdir_syscall_tests" section.
-    //
+    /// ### Function Arguments
+    /// 
+    /// The mkdir_syscall() receives two arguments:
+    /// 1. Path - This represents the path at which the new directory will be created.
+    ///           For example: "/parentdir/dir" represents the new directory name as "dir", which will be created at this path (/parentdir/dir).
```

Comment:
This may cause problems with the test runner.  It assumes that more than 4 spaces is code to execute.